src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+#include "ac_npc_shop_master.h"
+
+#include "m_shop.h"
+#include "m_common_data.h"
+#include "m_font.h"
+#include "m_msg.h"
+#include "m_private.h"
+#include "m_string.h"
+#include "m_actor_type.h"
+#include "m_player_lib.h"
+#include "m_mail_password_check.h"
+#include "m_actor.h"
+#include "m_actor_type.h"
+#include "m_camera2.h"
+#include "m_house.h"
+#include "m_melody.h"
+#include "m_scene.h"
+#include "m_collision_bg.h"
+#include "m_demo.h"
+#include "m_player.h"
+#include "m_bgm.h"
+#include "ac_npc.h"
+#include "ac_uki.h"
+#include "ac_npc_anim_def.h"
+
+static void aNSM_actor_ct(ACTOR *shop_master, GAME* game);
+static void aNSM_actor_dt(ACTOR* actorx, GAME* game);
+static void aNSM_actor_move(ACTOR* actorx, GAME* game);
+static void aNSM_actor_draw(ACTOR* actorx, GAME* game);
+static void aNSM_actor_init(ACTOR* actorx, GAME* game);
+static void aNSM_actor_save(ACTOR* actorx, GAME* game);
+
+static void aNSM_talk_request(ACTOR* actorx, GAME* game);
+static int aNSM_talk_init(ACTOR* actorx, GAME* game);
+static int aNSM_talk_end_chk(ACTOR* actorx, GAME* game);
+
+static void aNSC_BGcheck(ACTOR *actorx);
+static void aNSC_set_zone_data(NPC_SHOP_MASTER_ACTOR *shop_master, ACTOR *actor);
+static void aNSC_set_player_angl(NPC_SHOP_MASTER_ACTOR *shop_master);
+static void aNSC_talk_demo_proc(ACTOR *actorx);
+static void aNSC_sell_camera(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game);
+
+static void aNSM_actor_ct(ACTOR *actorx, GAME* game) {
+    static aNPC_ct_data_c ct_data = {
+        &aNSM_actor_move,
+        &aNSM_actor_draw,
+        0,
+        &aNSM_talk_request,
+        &aNSM_talk_init,
+        &aNSM_talk_end_chk,
+        TRUE 
+    };
+
+    NPC_SHOP_MASTER_ACTOR *shop_master = (NPC_SHOP_MASTER_ACTOR *)actorx;
+    int action = 0;
+
+    CLIP(npc_clip)->ct_proc(actorx, game, &ct_data);
+
+    shop_master->npc_class.draw.main_animation.keyframe.morph_counter = 0.0f;
+    shop_master->sell_item = EMPTY_NO;
+    shop_master->npc_class.condition_info.hide_flg = FALSE;
+    shop_master->_9C0 = -1;
+    actorx->shape_info.draw_shadow = TRUE;
+
+    if (Common_Get(door_data).door_actor_name == RSV_NO) {
+        shop_master->npc_class.talk_info.melody_inst = 0;
+        action = 0x3d; 
+    } else {
+        action = 0;
+    }
+
+    aNSC_setupAction(shop_master, (GAME_PLAY *)game, action);
+}
+
+static void aNSM_actor_save(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->save_proc)(actorx, game);
```

Comment:
Firstly, we now prefer calling clip function pointers without the explicit dereference. Secondly, we should use CLIP(n) instead of accessing Common_Get(clip) if possible.

This would look like:
`CLIP(npc_clip)->save_proc(actorx, game);`